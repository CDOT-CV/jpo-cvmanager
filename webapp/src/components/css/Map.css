.rsu-marker {
    margin-top: -6px;
    margin-left: -6px;
}

.legend {
    z-index: 90;
    background: #0e2052;
    margin: 10px;
    border-radius: 15px;
}

.legend-header {
    font-family: Arial, Helvetica, sans-serif;
    font-size: 24px;
    margin-top: 5px;
    margin-left: 10px;
    margin-right: 10px;
    color: white;
}

.legend-item {
    margin-top: 8px;
    margin-bottom: 8px;
    margin-left: 10px;
    margin-right: 10px;
}

.legend-label {
    font-family: Arial, Helvetica, sans-serif;
    font-size: 18px;
    color: white;
}

.legend-input {
    margin-right: 10px;
    accent-color: #d16d15;
}
