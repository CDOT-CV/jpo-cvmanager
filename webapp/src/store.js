import { configureStore } from "@reduxjs/toolkit";
import rsuReducer from "./slices/rsuSlice";
import userReducer from "./slices/userSlice";
import wzdxReducer from "./slices/wzdxSlice";
import configReducer from "./slices/configSlice";

const store = configureStore({
  reducer: {
    rsu: rsuReducer,
    user: userReducer,
    wzdx: wzdxReducer,
    config: configReducer,
  },
});

export default store;
