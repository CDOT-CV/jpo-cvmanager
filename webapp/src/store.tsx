import { Action, ThunkAction, configureStore } from '@reduxjs/toolkit'
import rsuReducer from './generalSlices/rsuSlice'
import userReducer from './generalSlices/userSlice'
import wzdxReducer from './generalSlices/wzdxSlice'
import configReducer from './generalSlices/configSlice'
import adminAddOrganizationReducer from './features/adminAddOrganization/adminAddOrganizationSlice'
import adminAddRsuReducer from './features/adminAddRsu/adminAddRsuSlice'
import adminAddUserReducer from './features/adminAdduser/adminAddUserSlice'
import adminEditOrganizationReducer from './features/adminEditOrganization/adminEditOrganizationSlice'
import adminEditRsuReducer from './features/adminEditRsu/adminEditRsuSlice'
import adminEditUserReducer from './features/adminEditUser/adminEditUserSlice'
import adminOrganizationTabReducer from './features/adminOrganizationTab/adminOrganizationTabSlice'
import adminOrganizationTabUserReducer from './features/adminOrganizationTabUser/adminOrganizationTabUserSlice'
import adminOrganizationTabRsuReducer from './features/adminOrganizationTabRsu/adminOrganizationTabRsuSlice'
import adminRsuTabReducer from './features/adminRsuTab/adminRsuTabSlice'
import adminUserTabReducer from './features/adminUserTab/adminUserTabSlice'
import menuReducer from './features/menu/menuSlice'
import intersectionMapReducer from './components/map/map-slice'
import intersectionMapLayerStyleReducer from './components/map/map-layer-style-slice'

export const setupStore = (preloadedState: any) => {
  return configureStore({
    reducer: {
      rsu: rsuReducer,
      user: userReducer,
      wzdx: wzdxReducer,
      config: configReducer,
      adminAddOrganization: adminAddOrganizationReducer,
      adminAddRsu: adminAddRsuReducer,
      adminAddUser: adminAddUserReducer,
      adminEditOrganization: adminEditOrganizationReducer,
      adminEditRsu: adminEditRsuReducer,
      adminEditUser: adminEditUserReducer,
      adminOrganizationTab: adminOrganizationTabReducer,
      adminOrganizationTabUser: adminOrganizationTabUserReducer,
      adminOrganizationTabRsu: adminOrganizationTabRsuReducer,
      adminRsuTab: adminRsuTabReducer,
      adminUserTab: adminUserTabReducer,
      menu: menuReducer,
      intersectionMap: intersectionMapReducer,
      intersectionMapLayerStyle: intersectionMapLayerStyleReducer,
    },
    preloadedState,
  })
}

type AppStore = ReturnType<typeof setupStore>
export type AppState = ReturnType<AppStore['getState']>
type AppThunk<ReturnType = void> = ThunkAction<ReturnType, AppState, unknown, Action>

export type RootState = ReturnType<ReturnType<typeof setupStore>['getState']>
