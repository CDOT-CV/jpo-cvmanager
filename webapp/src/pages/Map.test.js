import React from "react";
import { render } from "@testing-library/react";
import Map from "./Map";
import { Provider } from "react-redux";
import { setupStore } from "../store";

it("should take a snapshot", () => {
  const { asFragment } = render(
    <Provider store={setupStore({})}>
      <Map />
    </Provider>
  );

  expect(
    asFragment(
      <Provider store={setupStore({})}>
        <Map />
      </Provider>
    )
  ).toMatchSnapshot();
});
