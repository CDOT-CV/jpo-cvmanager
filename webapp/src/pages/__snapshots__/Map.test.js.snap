// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
      style="position: absolute;"
    >
      <button
        style="height: 35px; padding: 1px 12px; margin: 10px 10px; text-align: center; text-decoration: none; font-size: 18px; background: rgb(209, 109, 21); border-radius: 30px; cursor: pointer; color: white; z-index: 90;"
      >
        SCMS Status
      </button>
    </div>
    <div
      style="position: relative; width: 100%; height: calc(100vh - 100px); cursor: grab; outline: none;"
      tabindex="0"
    >
      <div
        style="position: relative; width: 100%; height: 100%;"
      >
        <div
          class=""
          style="position: absolute; width: 100%; height: 100%; overflow: hidden; visibility: inherit;"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
